src/node_mgmt/commands.py

```diff
@@ -211,6 +218,66 @@ def readall(s):
     return result
 
 
+def holykell_sensor_address_7(node):
+    return _change_address_holykell(original_slave_id=1, target_slave_id=7)
+
+
+def holykell_sensor_address_8(node):
+    return _change_address_holykell(original_slave_id=1, target_slave_id=8)
+
+
+def _change_address_holykell(original_slave_id: int, target_slave_id: int) -> dict:
+    mod = minimalmodbus.Instrument(port=DEFAULT_SERIAL_DEV, slaveaddress=original_slave_id, debug=True)
+    mod.serial.baudrate = DEFAULT_SERIAL_BAUD_RATE
+    mod.serial.timeout = 3
+    result = {}
+
+    try:
+        # check that no device is already on target slave address
+        _read_holykell(mod, slave_id=target_slave_id)
+    except minimalmodbus.NoResponseError:
+        # no device detected on target slave, go ahead and set address
+        result = _set_address_holykell(mod, result, original_slave_id=original_slave_id,
+                                       target_slave_id=target_slave_id)
+    except Exception as e:
+        # if any other exception than NoResponseError is caught, the command must fail
+        logger.warning(f'Failed to check if slave {target_slave_id} available. Exception {e}')
+        result['Error'] = f'Failed to check if slave {target_slave_id} available. Exception {e}'
+    else:
+        logger.info(f'Slave ID {target_slave_id} already in use')
+        result['Error'] = f'Other device already detected on slave {target_slave_id}'
+
+    return result
+
+
+def _set_address_holykell(mod: minimalmodbus.Instrument, result: dict,
+                          original_slave_id: int, target_slave_id: int) -> dict:
+    logger.info(f'Slave {target_slave_id} free, assigning the device to it')
+    result[f'Check on slave {target_slave_id}'] = 'Slave available, assigning the device to it'
+    try:
+        # command to change slave ID
+        mod.address = original_slave_id
+        mod.write_register(80, target_slave_id, 0, 6)
+        sleep(1)
+        # command to save changes
+        mod.address = target_slave_id
+        mod.write_register(64, 49087, 0, 6)
+        sleep(1)
+        # confirmation that data can be read after change
+        result[f'Success, fuel level read from slave {target_slave_id} (mm)'] = mod.read_registers(2, 1, 3)
+        logger.info(f'Holykell successfully assigned to slave ID {target_slave_id}')
+        return result
+    except Exception as e:
+        result['Error'] = f'Unable to assign slave ID to {target_slave_id}. Exception {e}'
+        logger.warning(f'Unable to assign slave ID to {target_slave_id}. Exception {e}')
+        return result
+
+
+def _read_holykell(mod, slave_id: int):
```

Comment:
Type annotation here also

Reply:
Thanks for double checking, this is corrected